src/main/java/com/gooddata/warehouse/WarehouseS3CredentialsList.java

```diff
@@ -0,0 +1,36 @@
+/*
+ * Copyright (C) 2007-2017, GoodData(R) Corporation. All rights reserved.
+ */
+package com.gooddata.warehouse;
+
+import com.fasterxml.jackson.annotation.JsonIgnoreProperties;
+import com.fasterxml.jackson.annotation.JsonTypeInfo;
+import com.fasterxml.jackson.annotation.JsonTypeName;
+import com.fasterxml.jackson.databind.annotation.JsonDeserialize;
+import com.fasterxml.jackson.databind.annotation.JsonSerialize;
+import com.gooddata.collections.PageableList;
+import org.springframework.web.util.UriTemplate;
+
+import java.util.List;
+import java.util.Map;
+
+@JsonTypeInfo(include = JsonTypeInfo.As.WRAPPER_OBJECT, use = JsonTypeInfo.Id.NAME)
+@JsonTypeName("s3CredentialsList")
+@JsonIgnoreProperties(ignoreUnknown = true)
+@JsonDeserialize(using = WarehouseS3CredentialsListDeserializer.class)
+@JsonSerialize(using = WarehouseS3CredentialsListSerializer.class)
+public class WarehouseS3CredentialsList extends PageableList<WarehouseS3Credentials> {
```

Comment:
ok I figured that since there is no change in visibility of `WS3CList` class, I have restored `URI` and `TEMPLATE` constants to their original state which is more consistent with other entities in this package.